packages/app/components/Today/QueueCreateButton.tsx

```diff
@@ -0,0 +1,146 @@
+  const [modalVisible, setModalVisible] = useState(false);
+  const { cid } = router.query;
+  const { course, mutateCourse } = useCourse(Number(cid));
+  const role = useRoleInCourse(Number(cid));
+  const queueCheckedIn = course?.queues.find((queue) =>
+    queue.staffList.find((staff) => staff.id === profile?.id)
+  );
+  const [form] = Form.useForm();
+
+  async function submitMakeQueue() {
+    const queueRequest = await form.validateFields();
+    try {
+      await API.taStatus.makeQueue(
+        Number(cid),
+        queueRequest.officeHourName,
+        !queueRequest.allowTA,
+        queueRequest.notes
+      );
+
+      message.success(
+        `created a new queue ${queueRequest.officeHourName}. Checking you in...`
+      );
+
+      const redirectID = await API.taStatus.checkIn(
+        Number(cid),
+        queueRequest.officeHourName
+      );
+
+      mutateCourse();
+
+      router.push(
+        "/course/[cid]/queue/[qid]",
+        `/course/${Number(cid)}/queue/${redirectID.id}`
+      );
+    } catch (err) {
+      message.error(err.response?.data?.message);
+    }
+  }
+
+  return (
+    <>
+      {modalVisible && (
+        <Modal
+          title={`Create a new queue`}
+          visible={modalVisible}
+          onCancel={() => {
+            setModalVisible(false);
+          }}
+          okText="Create"
+          onOk={submitMakeQueue}
+        >
+          <Form form={form}>
+            <Row>
+              <Col style={{ fontWeight: "bold" }}>
+                <Row>Online?</Row>
+                <Row>
+                  <Form.Item
+                    name="isOnline"
+                    initialValue={true}
+                    valuePropName="checked"
+                  >
+                    <Switch data-cy="qc-isonline" defaultChecked />
+                  </Form.Item>
+                </Row>
+              </Col>
+
+              <Col style={{ fontWeight: "bold" }}>
+                <Row>Allow TAs?</Row>
+                <Row>
+                  <Form.Item
+                    name="allowTA"
+                    initialValue={role === Role.TA}
+                    valuePropName="checked"
+                  >
+                    <Switch
+                      data-cy="qc-allowTA"
+                      disabled={role === Role.TA}
+                      defaultChecked={role === Role.TA}
+                    />
+                  </Form.Item>
+                </Row>
+              </Col>
+            </Row>
+
+            <Row style={{ fontWeight: "bold" }}>Queue Location</Row>
+
+            <Row>
+              <Form.Item
+                name="officeHourName"
+                rules={[
+                  {
+                    required: true,
+                    message: "Please give this room a name.",
+                  },
+                ]}
+                initialValue={
+                  role === Role.TA
+                    ? ""
+                    : `Professor ${profile.lastName}'s Office Hours`
```

Comment:
I think we probably don't need this initialValue for professors, since the idea is the make queues based off of room names anyways